### PR TITLE
Action inputs generation and retrieval implementation

### PR DESCRIPTION
**Pull request by spolu**
This PR implements:
- generic action inputs generation code path (could be used by other actions)
- partial retrieval action execution (generates the parameters, actual retrieval still TODO)

Associated Dust app for inputs generation is here: https://dust.tt/w/78bda07b39/a/f4816b1e13

Notes: I follow the error handling convention we agreed upon here. Result whenever we act on the error condition. Throwing means unrecoverable (but there is no throw here).

### CODE REVIEW

File: front/lib/api/assistant/actions/retrieval.ts [hunk 1]
```diff
@@ -1,33 +1,238 @@
...
+    if (d.dataSourceId.startsWith("managed-")) {
+      dataSourceName = d.dataSourceId.substring(8);
+    }
+
+    content += `TITLE: ${title} (data source: ${dataSourceName})\n`;
+    content += `REFERENCE: ${d.reference}\n`;
+
+    content += `EXTRACTS:\n`;
+    for (const c of d.chunks) {
+      content += `${c.text}\n`;
```
Comment (fontanierh) [inline front/lib/api/assistant/actions/retrieval.ts:98]: in the current form, there's nothing that tells the model about the chunk boundary right ? Or is the chunk content already stripped of all `\n` ?

File: front/lib/api/assistant/conversation.ts [hunk 2]
```diff
@@ -7,14 +7,133 @@ import {
...
+      return new Err(new Error(`Error tokenizing model message: ${res.error}`));
+    }
+
+    return new Ok(res.value.tokens.length);
+  }
+
+  // This is a bit aggressive but fuck it.
+  const tokenCountRes = await Promise.all(
+    messages.map((m) => {
+      return tokenCountForMessage(m, model);
```
Comment (fontanierh) [inline front/lib/api/assistant/conversation.ts:109]: let's monitor how it goes, but may be useful at some point to have a dedicated thread pool in `core` for tokenizing, right now it creates a new `spawn_blocking` thread each time which is suboptimal

File: front/types/assistant/actions/retrieval.ts [hunk 3]
```diff
@@ -4,6 +4,9 @@
 
 import { ModelId } from "@app/lib/databases";
 
+import { AgentActionConfigurationType } from "../agent";
```
Comment (fontanierh) [inline front/types/assistant/actions/retrieval.ts:7]: nit: relative import

**Review by fontanierh — DISMISSED** [review]
Looks good

File: front/lib/api/assistant/actions/retrieval.ts (hunk 1 above)
Comment (spolu) [reply front/lib/api/assistant/actions/retrieval.ts:98]: Chunks don't have `\n` :+1:

File: front/lib/api/assistant/conversation.ts (hunk 2 above)
Comment (spolu) [reply front/lib/api/assistant/conversation.ts:109]: Yes! We have 32 workers right now but definitely agree we should move it in its own thread pool :+1: 

Creating an issue on the Assistant board

Comment (spolu) [reply front/lib/api/assistant/conversation.ts:109]: https://github.com/dust-tt/dust/issues/1279

File: front/types/assistant/actions/retrieval.ts (hunk 3 above)
Comment (spolu) [reply front/types/assistant/actions/retrieval.ts]: Fixed :+1:

**Comment by spolu** [comment]
@PopDaph post-merge review welcome

**Pull request merged**